### PR TITLE
Better examples to show Compose file structure, expand/collapse accordion

### LINKED ISSUE #1349: Provide parent for each key in the compose file
File: [compose/compose-file.md](https://docs.docker.com/compose/compose-file/)

Into docker compose file reference, I would like to know the parent of each docker-compose property.
E.g.:
- 'build' property has 'serviceName' parent.
- 'services' has not parent (i.e. it's a root property)

### LINKED ISSUE #3955: Change link referring to Docker Swarms
File: [compose/overview.md](https://docs.docker.com/compose/overview/), CC @londoncalling 
In the section, Single host deployments, a reference is made to an "entire Docker Swarm cluster," where Docker Swarm is a link to https://docs.docker.com/swarm/overview/ . The page at that link location notes it is legacy documentation and that Swarm mode is integrated with Docker Engine. The new link is https://docs.docker.com/engine/swarm/ .

### PR DESCRIPTION
**Pull request by londoncalling**
### What's changed

- added more context to examples for `build` and `volume` sub-options to `service`
- added back in example Compose file at the top of the page to provide overall context
- fixed a link

### Related 

Fixes #1349 
Fixes #3955 

### Reviewers fyi

@shin- @MicheleMannino @Capricorn1

Signed-off-by: Victoria Bialas <victoria.bialas@docker.com>

### CODE REVIEW

File: compose/compose-file/index.md [hunk 1]
```diff
@@ -44,7 +40,115 @@ tutorial.
...
+Swarm](https://github.com/docker/labs/blob/master/beginner/chapters/votingapp.md).
+Click to show/hide the example Compose file below.
+
+<div class="panel-group" id="accordion" role="tablist" aria-multiselectable="true">
+
+  <div class="panel panel-default">
+    <div class="panel-heading" role="tab" id="headingThree">
+      <h5 class="panel-title" id="collapsible-group-item-3"> <a class="" role="button" data-toggle="collapse" data-parent="#accordion" data-target="#collapseThree" aria-expanded="true" aria-controls="collapseThree"> Example Compose file version 3 </a> </h5>
+    </div>
+    <div id="collapseThree" class="panel-collapse collapse" role="tabpanel" aria-labelledby="headingThree" aria-expanded="true">
```
Comment (joaofnfernandes) [inline compose/compose-file/index.md]: 💥  sweet!
Not a major issue, but the compose file includes 6 spaces, so when I copy this sample to my text editor I'll have to remove those spaces to make this sample work.
Try to remove the leading spaces in lines 58-146, that might do the trick

Comment (londoncalling) [inline compose/compose-file/index.md]: @joaofnfernandes _great_ suggestion, have a look now, should be good. (FYI, I'm going to look into the possibility of being able to pull this file from the voting app sample on GitHub in the long term.)

File: compose/compose-file/index.md [hunk 2]
```diff
@@ -44,7 +40,115 @@ tutorial.
...
+    ports:
+      - "8080:8080"
+    stop_grace_period: 1m30s
+    volumes:
+      - "/var/run/docker.sock:/var/run/docker.sock"
+    deploy:
+      placement:
+        constraints: [node.role == manager]
+
+  networks:
```
Comment (joaofnfernandes) [inline compose/compose-file/index.md]: Almost there. Network and volumes need to be aligned with services, so you need to remove 2 spaces from lines 141-146

**Comment by londoncalling** [comment]
@mstanleyjones @joaofnfernandes @johndmulhausen @shin- Please take a look if you get a chance. Accordion menu implemented for Compose sample config file, and instructions added in `test.md`. We can iterate and refine this more later, but I think it's quite workable for now.

File: _scss/_utilities.scss [hunk 3]
```diff
@@ -58,10 +58,37 @@ i.fa.fa-outdent {
...
+.panel-heading .chevron:after {
+    content: "\f0d8";
+}
+
+.panel-heading.collapsed .chevron:after {
+    content: "\f0d7";
+}
+
+.panel-heading:hover{
+    background: #fafdfe;
```
Comment (mstanleyjones) [inline _scss/_utilities.scss]: Can you please use `background-color`? `background` is a hybrid which can take a color and an image together. This is a nit but it's more correct.

**Comment by londoncalling** [comment]
LGTM with additional fixes to polish up the accordions, test/fix for mobile, and full explanation in `test.md`. Original goal was Compose file improvements per user feedback. Added accordion to support long code/config samples.

**Pull request merged**